pkgs/stdenv/adapters.nix

```diff
@@ -66,12 +66,10 @@ rec {
 
             # In nixpkgs, sometimes 'null' gets in as a buildInputs element,
             # and we handle that through isAttrs.
-            getNativeDrv = drv: drv.nativeDrv or drv;
-            getCrossDrv = drv: drv.crossDrv or drv;
-            nativeBuildInputsDrvs = map getNativeDrv nativeBuildInputs;
-            buildInputsDrvs = map getCrossDrv buildInputs;
-            propagatedBuildInputsDrvs = map getCrossDrv propagatedBuildInputs;
-            propagatedNativeBuildInputsDrvs = map getNativeDrv propagatedNativeBuildInputs;
+            nativeBuildInputsDrvs = nativeBuildInputs;
+            buildInputsDrvs = buildInputs;
+            propagatedBuildInputsDrvs = propagatedBuildInputs;
+            propagatedNativeBuildInputsDrvs = propagatedNativeBuildInputs;
```

Comment:
Ah, I did this as an intermediate step to break up diff, let me inline. In general, this adapter should shrink a lot / basically go away. Especially once ios becomes less of a special case 😛.